src/v/security/role.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "absl/container/flat_hash_set.h"
+#include "security/types.h"
+#include "serde/envelope.h"
+
+#include <seastar/core/sstring.hh>
+
+#include <iosfwd>
+
+namespace security {
+
+enum class role_member_type {
+    user = 0,
+};
+
+std::ostream& operator<<(std::ostream&, role_member_type);
+
+class role_member
+  : public serde::
+      envelope<role_member, serde::version<0>, serde::compat_version<0>> {
+public:
+    role_member() = default;
```

Comment:
This constructor is currently unused, and I'm not sure a default constructed role is meaningful.
```suggestion
```

Reply:
good point. done.

Reply:
Ah, looks like serde requires default-constructability. Adding it back.

(I don't see a way to circumvent that, but please lmk if I've missed something)

Reply:
Might work to make the default constructor private and private `friend std::is_default_constructible<role_member>`... building

Reply:
Nah, serde actually default constructs an instance for tag_invoke black magic. I guess a stern comment will have to do.

Reply:
I guess [this never made it](https://github.com/redpanda-data/redpanda/pull/1078#discussion_r626497175)

Reply:
there is at least one deep philosophical question on that thread.